src/main/messageExtractor/PeripheralAction.java

```diff
@@ -0,0 +1,24 @@
+package messageExtractor;
+
+/**
+ * Wrapper for the action extracted from the message sent to the server. This object is returned by the message
+ * extractor to be sent to the peripheral specified by the ID.
+ * @see MessageExtractorInterface
+ */
+public class PeripheralAction {
```

Comment:
This structure will be changed for the next release, so we will have a general class Message with subclasses action and request. A request is meant for the server, it will do things such as return a list of all peripherals. An action is always meant for a peripheral. This is the reason why we did not change the code so far.